public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,318 @@
+
+                                      if (newValueDetails.showExpression) {
+                                        setValue(
+                                          `config.transformations.${index}.expression`,
+                                          keptVals[index]?.expression || ''
+                                        );
+                                      } else {
+                                        setValue(`config.transformations.${index}.expression`, '');
+                                      }
+
+                                      if (newValueDetails.showMapValue) {
+                                        setValue(
+                                          `config.transformations.${index}.mapValue`,
+                                          keptVals[index]?.mapValue || ''
+                                        );
+                                      } else {
+                                        setValue(`config.transformations.${index}.mapValue`, '');
+                                      }
+
+                                      onChange(value.value);
+                                    }
+                                  }}
+                                  options={transformOptions}
+                                  width={25}
+                                  aria-label="Type"
+                                />
+                              );
+                            }}
+                            control={control}
+                            name={`config.transformations.${index}.type`}
+                            rules={{ required: { value: true, message: 'Please select a transformation type' } }}
+                          />
+                        </Field>
+                        <Field
+                          label={
+                            <Stack gap={0.5}>
+                              <Label>Field</Label>
+                              <Tooltip
+                                content={
+                                  <div>
+                                    <p>
+                                      Optional. The field to transform. If not specified, the transformation will be
+                                      applied to the results field.
+                                    </p>
+                                  </div>
+                                }
+                              >
+                                <Icon name="info-circle" size="sm" />
+                              </Tooltip>
+                            </Stack>
+                          }
+                        >
+                          <Input
+                            {...register(`config.transformations.${index}.field`)}
+                            readOnly={readOnly}
+                            defaultValue={fieldVal.field}
+                            label="field"
+                            aria-label="field"
```

Comment:
to pair with the above comment:
```suggestion
                            id={`config.transformations.${fieldVal.id}.field`}
```